src/jsx/components/Pagination/Search.jsx

```diff
@@ -0,0 +1,39 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+function Search({ query }) {
+  const [queryParams, setQueryParams] = useSearchParams();
+
+  const setQuery = (query) => {
+    queryParams.set('q', query);
+    setQueryParams(queryParams);
+  }
+
+  const handleSearch = (text) => {
+    if ([1, 2].includes(text.length) == false) {
```

Comment:
Creo que aquí podría ser más simple la condición. 
```suggestion
    if (text.length >= 1) {
```
Aunque de preferencia podría ser mejor usar la condición `>= 2`

Reply:
pero al usar `>= 2`  no haría la busqueda al borrar todo el texto, nos haría falta agregar un or